pynopegl-utils/pynopegl_utils/viewer/__init__.py

```diff
@@ -291,22 +291,35 @@ def _export_video(self, filename: str, res_index: int, profile_index: int, sampl
 
             fd_r, fd_w = os.pipe()
 
+            exe = ["ffmpeg"]
+            input = ["pipe:%d" % fd_r]
+
+            if platform.system() == "Windows":
+                import msvcrt
+
+                handle = msvcrt.get_osfhandle(fd_r)
+                os.set_handle_inheritable(handle, True)
+                input = [f"handle:{handle}"]
+                exec = [sys.executable, "-m", "pynopegl_utils.viewer.ffmpeg"]
+
             # fmt: off
-            cmd = [
-                "ffmpeg", "-r", "%d/%d" % fps,
+            cmd = exec + [
+                "-r", "%d/%d" % fps,
                 "-v", "warning",
                 "-nostats", "-nostdin",
                 "-f", "rawvideo",
                 "-video_size", "%dx%d" % (width, height),
                 "-pixel_format", "rgba",
-                "-i", "pipe:%d" % fd_r,
-            ] + profile.args + [
+                "-i"] + input + profile.args + [
```

Comment:
I don't think you need an array for the input

Reply:
Done.